gcc/rust/expand/rust-expand-visitor.cc

```diff
@@ -328,27 +328,27 @@ ExpandVisitor::expand_inner_stmts (AST::BlockExpr &expr)
 }
 
 void
-ExpandVisitor::maybe_expand_expr (std::unique_ptr<AST::Expr> &expr)
+ExpandVisitor::maybe_expand_expr (AST::Expr &expr)
 {
   expander.push_context (MacroExpander::ContextType::EXPR);
-  expr->accept_vis (*this);
+  expr.accept_vis (*this);
   expander.pop_context ();
 
   auto final_fragment = expander.take_expanded_fragment ();
   if (final_fragment.should_expand ()
       && final_fragment.is_expression_fragment ())
-    expr = final_fragment.take_expression_fragment ();
```

Comment:
It doesn't seem like this would work, since instances of derived classes of ```Expr``` can have different sizes

Reply:
You're right. I believe the only way out is to keep reference to the unique pointer for type/expr expansion.